packages/core/tocco-ui/src/Button/buttonStyle.js

```diff
@@ -63,13 +63,25 @@ export const buttonStyle = css`
   & > span:first-child {
     margin-left: 0;
   }
-  ${({icon}) =>
-    icon &&
-    `
-    @media screen and (max-width: 1024px) {
+  ${({icon, showIconOnly}) => {
+    const hideLabelStyles = `
       ${StyledLabelWrapper} {
         display: none;
       }
+    `
+
+    if (icon && showIconOnly) {
+      return hideLabelStyles
+    }
+
+    return (
+      icon &&
+      typeof showIconOnly === 'undefined' &&
```

Comment:
this line looks a bit confusing to me. Is there a difference if `showIconOnly` is `undefined`, `null` or `false` or would it all be the same? Checking for `undefined` explicitly makes me expect that `undefined` really should be handled differently than e.g. `false`, but I can't see why this would be the case?

Reply:
So when having a valid value (`true` | `false`) the icon visibility is set explicitly. When having no value (`undefined`) the icon visibility is not set explicitly and the default responsiveness behaviour is used

Reply:
I've changed the context value, so that it easier to read 👍